src/common/service/NetRemoteService.cxx

```diff
@@ -466,6 +467,39 @@ NetRemoteService::WifiAccessPointSetAuthenticationConfiguration([[maybe_unused]]
     return grpc::Status::OK;
 }
 
+using Microsoft::Net::Wifi::Dot11FrequencyBand;
+
+::grpc::Status
+NetRemoteService::WifiAccessPointSetFrequencyBands([[maybe_unused]] ::grpc::ServerContext* context, const ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetFrequencyBandsRequest* request, ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetFrequencyBandsResult* response)
+{
+    LOGD << std::format("Received WifiAccessPointSetFrequencyBands request for access point id {}", request->accesspointid());
+
+    WifiAccessPointOperationStatus status{};
+
+    auto frequencyBands = request->frequencybands();
+
+    if (frequencyBands.size() <= 1 && frequencyBands[0] == Dot11FrequencyBand::Dot11FrequencyBandUnknown) {
+        status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter);
+        status.set_message("No frequency band provided");
+    } else {

```

Comment:
This will perform an invalid access when size() == 0. Also, we might consider returning an error whenever the `Invalid` value is anywhere in the input. Also, please use `std::size()` as is done elsewhere in the project:

```suggestion
    if (std::empty(frequencyBands)) {
        status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter);
        status.set_message("No frequency band provided");
    } else if (std::ranges::find(frequencyBands, Dot11FrequencyBand::Dot11FrequencyBandUnknown) != std::cend(frequencyBands)) {
        status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter);
        status.set_message("Invalid frequency band provided");
    } else {
```

Reply:
Thanks, I meant for that first line to be a placeholder that I was going to come back to fix but I forgot to. Will update